lib/integrations/awsoidc/list_security_groups.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/gravitational/trace"
+)
+
+// ListSecurityGroupsRequest contains the required fields to list VPC Security Groups.
+type ListSecurityGroupsRequest struct {
+	// VPCID is the VPC to filter Security Groups.
+	VPCID string
+
+	// NextToken is the token to be used to fetch the next page.
+	// If empty, the first page is fetched.
+	NextToken string
+}
+
+// CheckAndSetDefaults checks if the required fields are present.
+func (req *ListSecurityGroupsRequest) CheckAndSetDefaults() error {
+	if req.VPCID == "" {
+		return trace.BadParameter("vpc id is required")
+	}
+
+	return nil
+}
+
+// SecurityGroup is the Teleport representation of an EC2 Instance Connect Endpoint
+type SecurityGroup struct {
+	// Name is the Security Group name.
+	// This is just a friendly name and should not be used for further API calls
+	Name string
+
+	// SecurityGroupID is the security group ID.
+	// This is the value that should be used when doing further API calls.
+	SecurityGroupID string
```

Comment:
Nit: I'd consider renaming this field to just `ID`. Like the name field above is just `Name`, not `SecurityGroupName`.
```suggestion
	ID string
```